### PR TITLE
support file configuration of  `signature_algorithm_suite`

### PR DESCRIPTION
**Pull request by nklaassen**
This is currently only configurable via the dynamic `cluster_auth_preference` resource, this PR makes it configurable in the config file as well. This is very useful for starting up new clusters and making sure the initial generated keys (including the CA keys) have the right algorithms.

Part of [RFD 136](https://github.com/gravitational/teleport/blob/master/rfd/0136-modern-signature-algorithms.md).

### CODE REVIEW

**Review by tigrato — APPROVED** [review]

File: lib/config/fileconf_test.go [hunk 1]
```diff
@@ -402,20 +394,82 @@ func TestAuthenticationSection(t *testing.T) {
...
 	}
 	for _, tt := range tests {
 		t.Run(tt.desc, func(t *testing.T) {
 			text := bytes.NewBuffer(editConfig(t, tt.mutate))
 			cfg, err := ReadConfig(text)
-			tt.expectError(t, err)
+			if tt.expectError != nil {
+				tt.expectError(t, err)
+				return
+			}
```
Comment (GavinFrazar) [inline lib/config/fileconf_test.go:473]: when expectError is nil it should require.NoError

**Review by GavinFrazar — APPROVED** [review]
what takes precedence if there is a conflict between the static setting vs dynamic setting?

**Comment by nklaassen** [comment]
> what takes precedence if there is a conflict between the static setting vs dynamic setting?

it's handled here for the entire auth preference. Basically if anything is set in the config file, it overwrites the dynamic configuration on restart

https://github.com/gravitational/teleport/blob/2f61d55ae8b86d603444c83c8bcc3fe939b6eae3/lib/auth/init.go#L918-L926
https://github.com/gravitational/teleport/blob/2f61d55ae8b86d603444c83c8bcc3fe939b6eae3/lib/config/configuration.go#L943-L945

File: lib/config/fileconf_test.go (hunk 1 above)
Comment (nklaassen) [reply lib/config/fileconf_test.go:474]: good catch thanks

**Pull request merged**